src/yunohost/data_migrations/0009_setup_group_permission.py

```diff
@@ -0,0 +1,142 @@
+        auth.add('cn=all_users,ou=groups', attr_dict)
+
+        for rdn, attr_dict in ldap_map['depends_children'].items():
+            auth.add(rdn, attr_dict)
+    except Exception as e:
+        raise YunohostError("migration_0009_LDAP_update_failed", error=e)
+
+    logger.info(m18n.n("migration_0009_create_group"))
+
+    #Create a group for each yunohost user
+    user_list = auth.search('ou=users,dc=yunohost,dc=org',
+                            '(&(objectclass=person)(!(uid=root))(!(uid=nobody)))',
+                            ['uid', 'uidNumber'])
+    for user_info in user_list:
+        username = user_info['uid'][0]
+        auth.update('uid=%s,ou=users' % username,
+                    {'objectClass': ['mailAccount', 'inetOrgPerson', 'posixAccount', 'userPermissionYnh']})
+        user_group_add(auth, username, gid=user_info['uidNumber'][0], sync_perm=False)
+        user_group_update(auth, groupname=username, add_user=username, force=True, sync_perm=False)
+        user_group_update(auth, 'all_users', add_user=username, force=True, sync_perm=False)
+
+
+def migrate_app_permission(auth, app=None):
+    logger.info(m18n.n("migration_0009_migrate_permission"))
+
+    if app:
+        apps = app_list(installed=True, filter=app)['apps']
+    else:
+        apps = app_list(installed=True)['apps']
+
+    for app_info in apps:
+        app = app_info['id']
+        permission = app_setting(app, 'allowed_users')
+        path = app_setting(app, 'path')
+        domain = app_setting(app, 'domain')
+
+        url = None
+        if domain and path:
+            url = domain + path
+        permission_add(auth, app, 'main', urls=[url], default_allow=True, sync_perm=False)
+        if permission:
+            allowed_group = permission.split(',')
+            user_permission_add(auth, [app], 'main', group=allowed_group, sync_perm=False)
+        app_setting(app, 'allowed_users', delete=True)
+
+
+class MyMigration(Migration):
+    """
+        Update the LDAP DB to be able to store the permission
+        Create a group for each yunohost user
+        Migrate app permission from apps setting to LDAP
+    """
+
+    required = True
+
+    def migrate(self):
+        # Backup LDAP and the apps settings before to do the migration
+        logger.info(m18n.n("migration_0009_backup_before_migration"))
+        try:
+            backup_folder = "/home/yunohost.backup/premigration/" + time.strftime('%Y%m%d-%H%M%S', time.gmtime())
+            os.makedirs(backup_folder, 0o750)
+            os.system("systemctl stop slapd")
+            os.system("cp -r --preserve /etc/ldap %s/ldap_config" % backup_folder)
+            os.system("cp -r --preserve /var/lib/ldap %s/ldap_db" % backup_folder)
+            os.system("cp -r --preserve /etc/yunohost/apps %s/apps_settings" % backup_folder)
+            os.system("systemctl start slapd")
+        except Exception as e:
+            raise YunohostError("migration_0009_can_not_backup_before_migration", error=e)
```

Comment:
I dunno if that's really needed, but maybe to be secure we might want to add a : 
```
   os.system("systemctl restart slapd")
```
in the exception block ?

Reply:
Yes, but what happens if `systemctl restart slapd` generate an exception too ?